File: octave/puntofijo.m
function p = puntofijo(g,p0,N, tol)
  i = 1;
  fprintf("Metodo de punto fijo\n")
  fprintf("    i    p    error\n")
  while i <= N
    p = g(p0);
    fprintf("   %4.0f   %4.5f    %4.6f\n",i,p,abs(p-p0))
    if abs(p-p0)<tol
      fprintf("se terminaron las iteraciones")
      break
    end
    i = i + 1;
    p0 = p;
  end
end

File: python/puntofijo.py
import math
from tabulate import tabulate

head = ["i", "p", "error"]
tableData = []
def puntofijo(p0, n, tol, g):
    i = 1
    while i <= n:
        p = g(p0)
        tableData.append([i, p, abs(p - p0)])
        if abs(p-p0) < tol:
            break
        i += 1
        p0 = p
    print(tabulate(tableData, headers = head, tablefmt = "grid"))


print("Si se van a usar funciones matematicas fuera del alcance de python en este archivo, usar math")
p0 = float(eval(input("ingrese la aproximaciòn inicial: ")))
n = int(input("numero de iteraciones: "))
tol = float(input("ingrese la tolerancia: "))
g = eval(input("ingrese la funcion de punto fijo en lambda, ejemplo, lambda x: math.cos(x) - x --> "))
puntofijo(p0, n, tol, g)
